platforms/core-configuration/model-core/src/integTest/groovy/org/gradle/api/provider/CollectionPropertyIntegrationTest.groovy

```diff
@@ -306,40 +305,44 @@ task thing {
     def "reports failure to set property value using incompatible type"() {
         given:
         buildFile << """
-task wrongValueTypeDsl {
+task wrongValueTypeDsl(type: MyTask) {
     doLast {
-        verify.prop = 123
+        prop = 123
```

Comment:
Original code reached out into another task from this task, leading to serialization errors.